File: csharp/Almondbread/Mandelbrot.cs
using System;
using System.Numerics;
using System.Linq;

namespace Almondbread
{
    internal sealed class Mandelbrot
    {
        private const int width = 80;
        private const int height = 20;

        private static void Main(string[] args)
        {
            EachPoint(width, height, (x, y, v) =>
            {
                if (x == 0)
                {
                    Console.WriteLine();
                }
                Console.Write(v < 255 ? "-" : "#");
            });
        }

        private static short EscapeTime(Complex cc)
        {
            Func<Complex, Complex, short, short> escapeTime = null;
            escapeTime = (c, z, step) =>
            {
                if (step > 256 || Complex.Abs(z) > 4)
                {
                    return step;
                }
                return escapeTime(c, z * z + c, (short)(step + 1));
            };
            return escapeTime(cc, Complex.Zero, 0);
        }

        internal static void EachPoint(int width, int height, Action<int, int, short> func)
        {
            var r = 3.0 / width;
            var i = 2.0 / height;
            foreach (var t in from y in Enumerable.Range(0, height)
                              from x in Enumerable.Range(0, width)
                              select new { ir = -1.0 + i * y, ii = y, rr = -2.0 + r * x, ri = x })
            {
                func(t.ri, t.ii, EscapeTime(new Complex(t.rr, t.ir)));
            }
        }
    }
}

File: java/src/main/java/almondbread/Mandelbrot.java
package almondbread;

final class Mandelbrot {
    private static final int width = 80, height = 20;

    public static final void main(final String... args) {
        eachPoint(width, height, new Action3IntIntShort() {
                public void apply(final int x, final int y, final short v) {
                    if (x == 0) {
                        System.out.println();
                    }
                    System.out.print(v < 255 ? "-" : "#");
                }
        });
        System.exit(0);
    }

    private static final short escapeTime(final Complex c) {
        return escapeTimeRecursive(c, Complex.ZERO, (short) 0);
    }

    private static final short escapeTimeRecursive(final Complex c, final Complex z, final short step) {
        if (step > 256 || z.abs > 4) {
            return step;
        }
        return escapeTimeRecursive(c, z.mul(z).add(c), (short) (step + 1));
    }

    static final void eachPoint(final int width, final int height, final Action3IntIntShort func) {
        double r = -2.0, i = -1.0;
        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                func.apply(x, y, escapeTime(new Complex(r, i)));
                r += 3.0 / width;
            }
            r = -2.0;
            i += 2.0 / height;
        }
    }
}
